internal/cephfs/mounter/kernel.go

```diff
@@ -27,13 +29,45 @@ import (
 const (
 	volumeMounterKernel = "kernel"
 	netDev              = "_netdev"
+	kernelModule        = "ceph"
 )
 
-type KernelMounter struct{}
+// testErrorf can be set by unit test for enhanced error reporting.
+var testErrorf = func(fmt string, args ...any) { /* do nothing */ }
 
-func mountKernel(ctx context.Context, mountPoint string, cr *util.Credentials, volOptions *store.VolumeOptions) error {
-	if err := execCommandErr(ctx, "modprobe", "ceph"); err != nil {
-		return err
+type KernelMounter interface {
+	Mount(
+		ctx context.Context,
+		mountPoint string,
+		cr *util.Credentials,
+		volOptions *store.VolumeOptions,
+	) error
+
+	Name() string
+}
+
+type kernelMounter struct {
+	// needsModprobe indicates that the ceph kernel module is not loaded in
+	// the kernel yet (or compiled into it)
+	needsModprobe bool
+}
+
+func NewKernelMounter() KernelMounter {
+	return &kernelMounter{
+		needsModprobe: !filesystemSupported(kernelModule),
+	}
+}
+
+func (m *kernelMounter) mountKernel(
+	ctx context.Context,
+	mountPoint string,
+	cr *util.Credentials,
+	volOptions *store.VolumeOptions,
+) error {
+	if m.needsModprobe {
+		if err := execCommandErr(ctx, "modprobe", kernelModule); err != nil {
+			return err
+		}
```

Comment:
i think we can optimize it  by moving this modeprobe to `NewKernelMounter` so that we run it only once when driver starts not always WDYT?

Reply:
There is a check to see if the driver is loaded when `NewKernelMounter()` is called. That sets `m.needsModprobe`. It is done so that `NewKernelMounter()` does not need to return an error, keeping other changes minimal.

Reply:
Ah, yes, if `modprobe` succeeded, it should be set to `true`